src/System.Net.Primitives/src/System/Net/NetworkCredential.cs

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Security;
+using System.Text;
+using System.Threading;
+using Microsoft.Win32;
+
+namespace System.Net
+{
+    /// <devdoc>
+    ///    <para>
+    ///       Provides credentials for password-based authentication schemes such as basic,
+    ///       digest, NTLM and Kerberos.
+    ///    </para>
+    /// </devdoc>
+    public class NetworkCredential : ICredentials, ICredentialsByHost
+    {
+        private static readonly object s_lockingObject = new object();
+        private string _domain;
+        private string _userName;
+        private SecureString _password;
+
+        public NetworkCredential()
+            : this(string.Empty, string.Empty, string.Empty)
+        {
+        }
+
+        /// <devdoc>
+        ///    <para>
+        ///       Initializes a new instance of the <see cref='System.Net.NetworkCredential'/>
+        ///       class with name and password set as specified.
+        ///    </para>
+        /// </devdoc>
+        public NetworkCredential(string userName, string password)
+        : this(userName, password, string.Empty)
+        {
+        }
+
+        /// <devdoc>
+        ///    <para>
+        ///       Initializes a new instance of the <see cref='System.Net.NetworkCredential'/>
+        ///       class with name and password set as specified.
+        ///    </para>
+        /// </devdoc>
+        public NetworkCredential(string userName, SecureString password)
```

Comment:
There are 2 aspects to this:

1.  The `SecureString` class is not available in NETNative (UWP) contracts.  And while it is currently available in CoreFX contracts, it is probably going away.  So, it is important that there are no public API surfaces in the `NetworkCredential` class that have `SecureString` in them.

2. As far as how/where to store the data inside the `NetworkCredential` object (implementation), you should NOT use `SecureString`.  The reality is that, yes, you will store it in memory as plaintext.  I.e. the format that it came from in the public constructor/property.